src/librustc_middle/hir/map/collector.rs

```diff
@@ -195,7 +200,7 @@ impl<'a, 'hir> NodeCollector<'a, 'hir> {
         (self.map, svh)
     }
 
-    fn insert_entry(&mut self, id: HirId, entry: Entry<'hir>, hash: Fingerprint) {
+    fn insert_entry(&mut self, span: Span, id: HirId, entry: Entry<'hir>, hash: Fingerprint) {
```

Comment:
hmm. Why does the `span` come before the `id` in this API? The `id` is the "key" in the map, right? So I would have the `span` come at some point after it in the parameter list, just like the other "payloads"

Reply:
I kept the API from the `insert` function. This span parameter is only used to assert the consistency of the span map. It will be removed later.